### ISSUE TITLE
ASCellNode setSelected: called multiple times while scrolling

### ISSUE THREAD

**Issue by hashemp206** [facebook/AsyncDisplayKit#807]
Here is details: using AsyncDisplayKit 1.9

1. **first issue:**  when I select a cell, ```setSelected:``` of target cell gets called triple times. but I have tapped on the target cell only once!

2. **second issue:** when I deselect the target cell, by tapping again on it, cell gets deselected, but when I scroll I see it gets selected again and it called multiple times with value ```YES```!!!

you can see the issue with provided log in the sample project 
here is [sample project](https://dl.dropboxusercontent.com/u/45698608/ASCollectionView%20With%20Selection%20issue.zip)

**Comment by levi** [comment]
I'm seeing this issue too. Looking into a fix for it this today.

**Comment by appleguy** [comment]
@vitalybaev - any input here?  This issue is pretty bad.  I am tempted to disable the feature until we can fix...

**Comment by appleguy** [comment]
Unclear if we will actually do a 1.9.2, but because I'd like to launch 1.9.1 tonight and this isn't yet fixed / yet this is a bug that is probably worth fixing before 2.0, I'll create the 1.9.2 milestone for now.

**Comment by vitalybaev** [comment]
@appleguy as I told you, I haven't any time for 4 days, and I'll look deeply into it after that.
When I tested this issue, actually setSelected gets called 2 times, cause ```cell.selected = YES;``` gets called in ```didSelectItemAtIndexPath```. That's all for now

**Comment by vitalybaev** [comment]
@appleguy could you email me to discuss this issue?
dj@vitalybaev.ru
thanks

**Comment by appleguy** [comment]
@hashemp206 could you re-try this in master, or in 1.9.2 when I release it in the next day or so?  I expected to launch 1.9.2 a little while ago, but am now back on schedule and will get it out soon (in part because I also need to land a diff that I /don't/ want in 1.9.2, but do want to get testing on in master).

If you still have an issue, please reopen this task - or comment in a new task that I just filed: https://github.com/facebook/AsyncDisplayKit/issues/892
